paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -163,6 +163,14 @@ class PaparazziPlugin : Plugin<Project> {
         test.outputs.dir(snapshotOutputDir)
 
         val paparazziProperties = project.properties.filterKeys { it.startsWith("app.cash.paparazzi") }
+        test.systemProperties.putAll(paparazziProperties)
+
+        // Explicitly register these as inputs so that they are considered when determining up-to-date.
+        // They have to be added later to test.systemProperties,
+        // because at this point in the configuration, the properties are not resolvable yet.
+        // They become resolvable after the last afterEvaluate runs.
+        test.inputs.property("paparazzi.test.record", isRecordRun)
+        test.inputs.property("paparazzi.test.verify", isVerifyRun)
```

Comment:
note to future us: should probably move this to an enum type, since these values are mutually exclusive to each other.

Reply:
Yeah, I had a similar thought, however mind you that the calculation of it could be very complicated because of the laziness needed in "when graph ready". Anyway, it might be doable.